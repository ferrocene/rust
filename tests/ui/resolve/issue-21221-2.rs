//@ revisions: edition2015 edition2018
//@ [edition2015] edition: 2015
//@ [edition2018] edition: 2018..

pub mod foo {
    pub mod bar {
        // note: trait T is not public, but being in the current
        // crate, it's fine to show it, since the programmer can
        // decide to make it public based on the suggestion ...
        pub trait T {}
    }
    // imports should be ignored:
    use self::bar::T;
}

pub mod baz {
    pub use foo;
    //[edition2018]~^ ERROR unresolved import `foo`
    pub use std::ops::{Mul as T};
}

struct Foo;
impl T for Foo { }
//~^ ERROR cannot find trait `T`

fn main() {}
