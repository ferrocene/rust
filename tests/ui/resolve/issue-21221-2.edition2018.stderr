error[E0432]: unresolved import `foo`
  --> $DIR/issue-21221-2.rs:17:13
   |
LL |     pub use foo;
   |             ^^^ no external crate `foo`
   |
help: consider importing this module instead
   |
LL |     pub use crate::foo;
   |             +++++++

error[E0405]: cannot find trait `T` in this scope
  --> $DIR/issue-21221-2.rs:23:6
   |
LL | impl T for Foo { }
   |      ^ not found in this scope
   |
help: consider importing one of these traits
   |
LL + use crate::baz::T;
   |
LL + use crate::foo::bar::T;
   |

error: aborting due to 2 previous errors

Some errors have detailed explanations: E0405, E0432.
For more information about an error, try `rustc --explain E0405`.
